### DISCUSSION TITLE
Teksty 'tooltipy' do kart

### DISCUSSION THREAD

**Discussion by emkarcinos** [emkarcinos/WMIAdventure#402] [Codding]
Znalazłem jeden podstawowy problem dotyczący kart, aż się dziwię że wcześniej tego nie dostrzegliśmy.

Jak wszyscy wiecie, karty mają poziomy i na każdym poziomie karta ma inne efekty, inne wartości. Przy planowaniu edytora założyliśmy, że karta ma jeden *tooltip* opisujący działanie karty. Tutaj twórcy mogą napisać w dowolny sposób, co robi ta karta, razem z wartościami mocy efektów. Problem pojawia się wtedy, gdy dokładamy poziomy, wtedy *tooltip* przestaje być aktualny.

Musimy ratować sytuację.

W dyskusji pod #351 @mattyl006 piszę o dodatkowym *tekściku* do karty. Wpadłem na taki pomysł, żeby ten tekścik to był obecny *tooltip* (nazwę można uzgodnić), bo jest on wspólny dla każdego poziomu karty. Musimy zatem dodać nowy opis do karty, który będzie inny w każdym poziomie.

Robię dyskusję na ten temat, bo sporo rzeczy może takie rozwiązanie zmienić i na pewno dojdą kolejne, nieplanowane zadania do tego sprintu. Rozważcie także sprawę organizacyjną, czy dalibyśmy radę zaimplementować sugerowane przez nas rozwiązania.

**Comment by mattyl006** [comment]
To może te opisy efektowe będą się generować automatycznie na podstawie efektów?

**Reply by emkarcinos** [reply]
Spoko pomysł, nie wiem tylko czy będzie się dało to generowanie zrobić tak, żeby ładnie się to czytało, ale do zrobienia jak najbardziej

**Reply by emkarcinos** [reply]
To rozwiązanie w sumie nie wpłynęłoby za bardzo na wygląd edytora kart ani na jego reaktywność, trzebaby tylko zaktualizować podgląd karty i requesta. Na backendzie byłoby więcej pracy, ale backend akurat mało robi w tym sprincie, więc to jest dobre rozwiązanie, które nie rozwali nam za bardzo sprinta.

**Reply by itchystone** [reply]
Ja w sumie od początku myślałem, że tooltip to jest ten luźny opis karty, a opisy słowne efektów będą generowane. Na większości kart jakie widziałem jest osobno opis efektów co robi karta, a pod spodem jest jakiś "luźny" opis fabularny, czy coś w tym stylu

Sugerowałem to już w tym draftowym designie karty:
![image](https://user-images.githubusercontent.com/43492764/126613413-54da5c46-e16f-4540-84e3-b1d156c66e32.png)

**Reply by itchystone** [reply]
Chyba to generowanie nie będzie takie trudne, np. dla karty o efektach:
- 15-25 dmg przeciwnik
- zamiana kolejności kart przeciwnik
- 7.5-12.5 leczenie gracz

wygenerowany mógłby być tekst:
Przeciwnik:
- zadaje 15.0 - 25.0 obrażeń
- zamienia kolejność kart

Posiadacz karty:
- leczy 7.5 - 12.5 hp

**Reply by emkarcinos** [reply]
Nawet możnaby zrobić jakiś ładny generator, który doda przecinki i kropki, jakieś spójniki tam gdzie trzeba.

**Reply by itchystone** [reply]
Można będzie się pobawić ^^

**Comment by emkarcinos** [comment]
Zrobiłem na to już issues: #411 #412 #413 

Jak je tworzyłem to pojawił mi się w głowie taki problem:
W edytorze kart będziemy chcieli podejrzeć jak wygląda opis na naszej karcie.

Jeżeli to serwer miałby dawać nam opis takiej karty, to musielibyśmy zrobić jakiś nowy widok do API, który dla danych efektów odpowie nam wygenerowaną treścią opisu. To trochę komplikuje sprawę, bo nie mamy za bardzo kontroli nad tą treścią, jeżeli wysyłamy już request z pełną kartą do serwera to ta treść może być dowolnie zmieniona (jakimś cudem). Dodatkowo wydaje mi się to strasznie przekombinowane rozwiązanie. Macie jakieś inne pomysły jak to zrobić?

**Reply by itchystone** [reply]
No faktycznie jest z tym trochę kombinacji, ale nie widzę za bardzo innej możliwości. Aby zapobieć niechcianej zmianie może serwer przy dostaniu requesta jeszcze raz wygeneruje opis i sprawdzi, czy są identyczne? Jeśli nie będą to odrzuci request

**Reply by emkarcinos** [reply]
Czyli sugerujesz żeby zrobić nowy widok do API, który zwróci nam opis efektów. A jaki request do niego wysyłamy? Listę efektów?

**Reply by itchystone** [reply]
Chyba tak będzie git
